cocotb/drivers/amba.py

```diff
@@ -129,21 +239,32 @@ def write(
             The write response value.
 
         Raises:
+            TestError: If any of the input parameters is invalid.
             AXIProtocolError: If write response from AXI is not ``OKAY``.
         """
+
+        try:
+            iter(value)
+        except TypeError:
+            value = (value,)    # If value is not an iterable, make it
+
+        AXI4Master._check_burst(burst)
```

Comment:
This would become `burst = AXI4Burst(burst)`

Reply:
Should `burst` (the one passed by the user) be an enum or a str?

Reply:
I'd lean towards enum